src/sasctl/pzmm/write_json_files.py

```diff
@@ -950,3 +982,629 @@ def convertDataRole(self, dataRole):
             conversion = 1
 
         return conversion
+
+    @classmethod
+    def create_requirements_json(cls, json_path=Path.cwd()):
+        """
+        Searches the model directory for Python scripts and pickle files and determines
+        their Python package dependencies. Found dependencies are then matched to the package
+        version found in the current working environment. Then the package and version are
+        written to a requirements.json file.
+
+        WARNING:
+        The methods utilized in this function can determine package dependencies from provided
+        scripts and pickle files, but CANNOT determine the required package versions without
+        being in the development environment which they were originally created.
+
+        This function works best when run in the model development environment and is likely to
+        throw errors if run in another environment (and/or produce incorrect package versions).
+        In the case of using this function outside the model development environment, it is
+        recommended to the user that they adjust the requirements.json file's package versions
+        to match the model development environment.
+
+        Parameters
+        ----------
+        json_path : str, optional
+            The path to a Python project, by default the current working directory.
+
+        Yields
```

Comment:
Does not yield anything.

Currently returns None - is there a use-case for returning the JSON string (or dict) or do results always need to be written to disk?

Reply:
I could definitely see a use case for determining the development environment in the form of json or dict return. Is create_requirements_json() a broad enough function name to still work for that situation?

Reply:
sounds good to me

Reply:
Committed changes (unpushed currently) that handle the json writing in a cleaner, more pythonic way and included a return of the list of dicts that represents the information in the requirements.json file.